observability/src/main/java/io/grpc/observability/interceptors/InternalLoggingServerInterceptor.java

```diff
@@ -16,32 +16,224 @@
+        : Durations.fromNanos(deadline.timeRemaining(TimeUnit.NANOSECONDS));
+
+    // TODO (dnvindhya): implement isMethodToBeLogged() to check for methods to be logged
+    // according to config. Until then always return true.
+    if (!helper.isMethodToBeLogged(call.getMethodDescriptor().getFullMethodName())) {
+      return next.startCall(call, headers);
+    }
+
+    // Event: EventType.GRPC_CALL_REQUEST_HEADER
+    try {
+      helper.logRequestHeader(
+          seq.getAndIncrement(),
+          serviceName,
+          methodName,
+          authority,
+          timeout,
+          headers,
+          EventLogger.LOGGER_SERVER,
+          rpcId,
+          peerAddress);
+    } catch (Exception e) {
+      // Catching generic exceptions instead of specific ones for all the events.
+      // This way we can catch both expected and unexpected exceptions instead of re-throwing
+      // exceptions to callers which will lead to RPC getting aborted.
+      // Expected exceptions to be caught:
+      // 1. IllegalArgumentException
+      // 2. NullPointerException
+      logger.log(Level.SEVERE, "Unable to log request header", e);
+    }
+
+    ServerCall<ReqT, RespT> wrapperCall =
+        new SimpleForwardingServerCall<ReqT, RespT>(call) {
+          @Override
+          public void sendMessage(RespT message) {
+            // Event: EventType.GRPC_CALL_RESPONSE_MESSAGE
+            try {
+              helper.logRpcMessage(
+                  seq.getAndIncrement(),
+                  serviceName,
+                  methodName,
+                  EventType.GRPC_CALL_RESPONSE_MESSAGE,
+                  message,
+                  EventLogger.LOGGER_SERVER,
+                  rpcId);
+            } catch (Exception e) {
+              logger.log(Level.SEVERE, "Unable to log response message", e);
+            }
+            super.sendMessage(message);
+          }
+
+          @Override
+          public void sendHeaders(Metadata headers) {
+            // Event: EventType.GRPC_CALL_RESPONSE_HEADER
+            try {
+              helper.logResponseHeader(
+                  seq.getAndIncrement(),
+                  serviceName,
+                  methodName,
+                  headers,
+                  EventLogger.LOGGER_SERVER,
+                  rpcId,
+                  null);
+            } catch (Exception e) {
+              logger.log(Level.SEVERE, "Unable to log response header", e);
+            }
+            super.sendHeaders(headers);
+          }
+
+          @Override
+          public void close(Status status, Metadata trailers) {
+            // Event: EventType.GRPC_CALL_TRAILER
+            try {
+              helper.logTrailer(
+                  seq.getAndIncrement(),
+                  serviceName,
+                  methodName,
+                  status,
+                  trailers,
+                  EventLogger.LOGGER_SERVER,
+                  rpcId,
+                  null);
+            } catch (Exception e) {
```

Comment:
I wonder if putting this catch clause and the `logger.log` statement should be moved into the corresponding `helper.log*` methods. This will make the caller side code simpler. This needs to be done for both client and server so we can do this in another PR